internal/ui/hide_console_windows.go

```diff
@@ -14,17 +14,95 @@
 
 package ui
 
-import "syscall"
+import (
+	"syscall"
+	"unsafe"
+)
+
+// hideConsoleWindowOnWindows will hide the console window that is showing when
+// compiling on Windows without specifying the '-ldflags "-Hwindowsgui"' flag.
+//
+// The hiding is done if the console window exists for less than 1 second
+// because during development you will want to run from the command line and it
+// is not supposed to be hidden whenever you 'go run' or 'go test' the program.
+// That is why the work-around is to assume that the developer console is always
+// open for >= 1 sec begore running the program. The end user will double-click
+// the executable, it will pop up a console window and then call this function
+// right away so the time between creating the console and running this function
+// should never surpass 1 second. In that case the console is then hidden.
+func hideConsoleWindowOnWindows() {
+	consoleWindow, _, _ := getConsoleWindow.Call()
+	if consoleWindow == 0 {
+		return // no console attached
+	}
+
+	var consoleProcessID uint32
+	getWindowThreadProcessId.Call(
+		consoleWindow,
+		uintptr(unsafe.Pointer(&consoleProcessID)),
+	)
+	if consoleProcessID == 0 {
+		return // error retrieving the console process ID
+	}
+
+	const (
+		PROCESS_QUERY_INFORMATION = 0x0400
+		FALSE                     = 0
+	)
+	consoleProcess, _, _ := openProcess.Call(
+		PROCESS_QUERY_INFORMATION,
+		FALSE,
+		uintptr(consoleProcessID),
+	)
+	if consoleProcess == 0 {
+		return // error retrieving the console process handle
```

Comment:
If this is unexpected, return error value (e.g. `errors.New("ui: OpenProcess failed")` is fine)